src/types/state.ts

```diff
@@ -42,6 +42,12 @@ export type SearchState = {
   inputHelperText: string;
 };
 
+// contains the indices of currently checked revisions
+// in searchResults state
+export type CheckedRevisionsState = {
+  revisions: number[];
```

Comment:
should this be 
```suggestion
  revisions: Revision[];
```

Reply:
This state holds the index of a revision in searchResults. I can rename this to revisionIndices to make this more clear.